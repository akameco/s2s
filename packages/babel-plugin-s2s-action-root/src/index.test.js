// @flow
import path from 'path'

import pluginTester from 'babel-plugin-tester'
import plugin from '.'

const cwd = path.resolve(__dirname, '..')
const filename = path.resolve(cwd, 'src', 'app', 'actionTypes.js')
const output = path.resolve(cwd, 'src', 'types', 'actions.js')

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename, configFile: false },
  pluginOptions: {
    input: 'src/__fixtures__/**/*.js',
    output,
    globOptions: { cwd },
  },
  tests: [
    {
      title: 'options',
      code: `// @flow`,
    },
  ],
})

pluginTester({
  plugin,
  babelOptions: { filename, configFile: false },
  tests: [
    {
      title: 'error',
      code: `// throw error`,
      error: /require input option/,
    },
  ],
})

pluginTester({
  plugin,
  babelOptions: { filename, configFile: false },
  pluginOptions: {
    input: 'src/__fixtures__/**/*.js',
  },
  tests: [
    {
      title: 'error',
      code: `// throw error`,
      error: /require output option/,
    },
  ],
})

pluginTester({
  plugin,
  babelOptions: { filename, configFile: false },
  snapshot: true,
  pluginOptions: {
    input: 'src/**/actionTypes.js',
    output,
    globOptions: { cwd },
  },
  tests: [
    {
      title: 'glob options',
      code: `// show options`,
    },
  ],
})
