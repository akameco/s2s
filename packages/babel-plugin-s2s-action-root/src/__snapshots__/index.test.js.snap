// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`s2s-action-root glob options: glob options 1`] = `
"
// show options

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
// show options
import type { Action as AppAction } from \\"../__fixtures__/app/actionTypes\\";
import type { Action as BobAction } from \\"../__fixtures__/bob/actionTypes\\";

export type ReduxInitAction = {
  type: '@@INIT'
};

export type Action = ReduxInitAction | AppAction | BobAction;
"
`;

exports[`s2s-action-root options: options 1`] = `
"
// @flow

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import type { Action as AppAction } from \\"../__fixtures__/app/actionTypes\\";
import type { Action as BobAction } from \\"../__fixtures__/bob/actionTypes\\";

export type ReduxInitAction = {
  type: '@@INIT'
};

export type Action = ReduxInitAction | AppAction | BobAction;
"
`;
